fwutil/lib.py

```diff
@@ -556,42 +554,45 @@ def get_status(self, force):
             for chassis_component_name, chassis_component in chassis_component_map.items():
                 component = self.__pcp.chassis_component_map[chassis_name][chassis_component_name]
 
-                firmware_path = NA
-                firmware_version_current = chassis_component.get_firmware_version()
-                firmware_version = firmware_version_current
-
-                status = self.FW_STATUS_UP_TO_DATE
-                info = NA
-
                 if component:
                     firmware_path = component[self.__pcp.FIRMWARE_KEY]
-                    firmware_version_available = component[self.__pcp.VERSION_KEY]
-                    firmware_version = "{} / {}".format(firmware_version_current, firmware_version_available)
-                    info = component[self.__pcp.INFO_KEY]
 
                     if self.__root_path is not None:
                         firmware_path = self.__root_path + firmware_path
 
-                    if force or firmware_version_current != firmware_version_available:
+                    firmware_version_current = chassis_component.get_firmware_version()
+
+                    if self.__pcp.VERSION_KEY in component:
+                        firmware_version_available = component[self.__pcp.VERSION_KEY]
+                    else:
+                        firmware_version_available = chassis_component.get_available_firmware_version(firmware_path)
+
+                    if self.__root_path is not None:
+                        firmware_path = component[self.__pcp.FIRMWARE_KEY]
+
+                    firmware_version = "{} / {}".format(firmware_version_current, firmware_version_available)
+
+                    if firmware_version_current != firmware_version_available:
                         status = self.FW_STATUS_UPDATE_REQUIRED
+                    else:
+                        status = self.FW_STATUS_UP_TO_DATE
 
-                status_table.append(
-                    [
-                        chassis_name if append_chassis_name else EMPTY,
-                        module_name if append_module_na else EMPTY,
-                        chassis_component_name,
-                        firmware_path,
-                        firmware_version,
-                        status,
-                        info
-                    ]
-                )
+                    status_table.append(
+                        [
+                            chassis_name if append_chassis_name else EMPTY,
+                            module_name if append_module_na else EMPTY,
```

Comment:
the na in the same is confusing. should we have it as append_module_name?

Reply:
@liat-grozovik in this case we have only two values: 1. N/A 2. Empty. Both serve as module_name. I can change it to be `append_module_name` - no issues if you think it will be more clear.